dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3160,6 +3163,22 @@ Sedp::signal_liveliness_unsecure(DDS::LivelinessQosPolicyKind kind)
   }
 }
 
+
+bool Sedp::send_type_lookup_request(XTypes::TypeIdentifierSeq& type_ids,
+                                    const DCPS::RepoId& reader)
+{
+  // TLS_TODO: verify reader
```

Comment:
Like that the guid is a reader? I'm not sure that's necessary here, but you can use `GuidConverter(reader).isReader()` to do that.